docs/auth.md

```diff
@@ -0,0 +1,287 @@
+often use client certificate authentication.
+
+By default, a main Kubernetes API server configured with the
+`--client-ca-file` option automatically creates a ConfigMap called
+`extension-apiserver-authentication` in the `kube-system` namespace,
+populated with the the client CA file.  The service catalog API server use
+this CA certificate as the CA used to verify client authentication. This
+way, client certificate users who can authenticate with the main
+Kubernetes system can also authenticate with the service catalog API
+server.
+
+See the [delegated token authentication](#delegated-token-authentication)
+section for more information about how the service catalog API server
+contacts the main Kubernetes API server to access this ConfigMap.
+
+If you wish to use a different client CA certificate to verify client
+certificate authentication, you can manually pass the `--client-ca-file`
+option to the service catalog API server.
+
+See the [x509 client
+certificates](https://kubernetes.io/docs/admin/authentication/#x509-client-certs)
+section of the Kubernetes documentation for more information.
+
+### Delegated Token Authentication
+
+Delegated token authentication authenticates clients who pass in a token
+using the `Authorization: Bearer $TOKEN` HTTP header.  This is the common
+authentication method used by most human Kubernetes clients, as well as
+in-cluster non-human clients.
+
+In this case, the service catalog API server extracts the token from the
+HTTP request, and verifies it against another API server. In common cases,
+this is the main Kubernetes API server.  This allows users who are can
+authentication with the main Kubernetes system to also authenticate with
+the service catalog API server.
+
+By default, the service catalog API server searches for the connection
+information and credentials that are automatically injected into every pod
+running on a Kubernetes cluster.
+
+If you do not wish to have the service catalog API server authenticate
+against the same cluster that it is running on, or if it is running
+outside of a cluster, you can pass the `--authentication-kubeconfig`
+option to the serice catalog API server to specify a different Kubeconfig
+file to use to connect.
+
+The [Webhook token
+authentication](https://kubernetes.io/docs/admin/authentication/#webhook-token-authentication)
+method described in the Kubernetes authentication documentation works
+similarly in principal to delegated token authentication, except that we
+use an existing Kubernetes cluster instead of an external webhook.
+
+### RequestHeader Authentication
+
+RequestHeader authentication authenticates connections from API server
+proxies, which themselves have already authenticated the client.  It works
+similarly to [client certificate
+authentication](#client-certificate-authentication): it validates the
+verifies the certificate of the proxy using a CA certificate.  However, it
```

Comment:
...it validates the verifies the certificate... sounds odd.
...it validates the certificate?